### DISCUSSION TITLE
Way to change effects in real time

### DISCUSSION THREAD

**Discussion by zoyt** [Quefumas/gensound#25] [Q&A]
Hi there,

Really enjoying playing with your library! Super impressed by the quality of documentation and energy you have for this project. I hope it goes super far!

I've started wondering if there was a way to marry your work another project of mine, which tracks hand joint angles in real time using OpenCV and Mediapipe, and use the values that get updated every while cycle to effect audio in some way. In other words, I have an array that updates as fast as my computer can handle, and I think it would be interesting to use those values as inputs into some kind of synth. Is this something that is possible within gensound?

Thank you!

**Comment by zoyt** [comment]
Upon further research, it looks like Pyo may actually be closer to what I'm looking for for this project, but will have to come back to Gensound sometime soon! 

Thanks!

**Comment by Quefumas** [comment]
Hello,

Thanks for your feedback! It gives a lot of motivation seeing that people are actually using Gensound for anything at all.

I agree that the current incarnation of Gensound is not very well-suited to your idea. It was a conscious design choice to allow only offline processing & generation, and it has its drawbacks, but also some benefits. I occasionally toy with the idea of making a parallel online version, but this would require significant changes to the core of the library, so without outside help I will not try to do that until I feel the current design has been more or less perfected.

One thing you could try is to use Gensound to pre-generate several audio streams, and manually mix them on the fly according to the user input, but that won't really let you use much of the flexibility available in Gensound.

Anyway, feel free to drop me a note and share ideas, bugs, or even code that generates music just for fun.
